testnet_bitcoin_wallet/segwit.py

```diff
@@ -0,0 +1,33 @@
+from ProgrammingBitcoin.tx import Tx
+from ProgrammingBitcoin.script import Script
+
+from bech32 import decode
+
+def make_p2wpkh_script(h160):
+    return Script([0x00, h160])
+
+def decode_bech32(addr, testnet=False):
+    if testnet:
+        hrp = "tb"
+    else:
+        hrp = "bc"
+    version, decoded = decode(hrp, addr)
+    if version == None:
+        raise ValueError("invalid address")
+    return(bytes.__new__(bytes, decoded))
```

Comment:
I suspect that you may want to return `version` here as well.

Reply:
Yes, when I implement sending and receiving using segwit version 1 addresses that will be helpful.